crates/router/src/types.rs

```diff
@@ -299,6 +314,29 @@ impl ErrorResponse {
     }
 }
 
+impl From<ConnectorAuthType> for AccessTokenRequestData {
+    fn from(connector_auth: ConnectorAuthType) -> Self {
+        match connector_auth {
+            ConnectorAuthType::HeaderKey { api_key } => Self {
+                app_id: api_key,
+                id: None,
+            },
+            ConnectorAuthType::BodyKey { api_key, key1 } => Self {
+                app_id: api_key,
+                id: Some(key1),
+            },
+            ConnectorAuthType::SignatureKey { api_key, key1, .. } => Self {
+                app_id: api_key,
+                id: Some(key1),
+            },
+            ConnectorAuthType::NoKey => Self {
```

Comment:
when this case will happen?

Reply:
This case will not happen and should not happen, should this be a try_from instead of from and raise an error?

Reply:
Handle as a error